sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -830,18 +861,27 @@ async def replace_throughput(
         :returns: ThroughputProperties for the container, updated with new throughput.
         :rtype: ~azure.cosmos.offer.ThroughputProperties
         """
-        properties = await self._get_properties()
-        link = properties["_self"]
-        query_spec = {
-            "query": "SELECT * FROM root r WHERE r.resource=@link",
-            "parameters": [{"name": "@link", "value": link}],
-        }
-        throughput_properties = [throughput async for throughput in
-                                 self.client_connection.QueryOffers(query_spec, **kwargs)]
-        if len(throughput_properties) == 0:
-            raise CosmosResourceNotFoundError(
-                status_code=StatusCodes.NOT_FOUND,
-                message="Could not find Offer for container " + self.container_link)
+        throughput_properties: List[Dict[str, Any]]
+        # In case of a container recreate we will run this a second time
+        for attempt in range(2):
+            if attempt == 1:
+                # Refresh cache if no offer throughput was received the first time
+                await self.read()
+            properties = await self._get_properties()
+            link = properties["_self"]
+            query_spec = {
+                "query": "SELECT * FROM root r WHERE r.resource=@link",
+                "parameters": [{"name": "@link", "value": link}],
+            }
+            throughput_properties = [throughput async for throughput in
+                                     self.client_connection.QueryOffers(query_spec, **kwargs)]
+            if len(throughput_properties) == 0:
+                if attempt < 1:
+                    continue
+                raise CosmosResourceNotFoundError(
```

Comment:
This logic predates the container recreate, and someone should correct me if I am wrong, but with readd offer it actually returns empty instead of erroring out when it is not found, so the error is raised manually so it behaves as other sdks.